server/app/views/applicant/AddressCorrectionBlockView.java

```diff
@@ -0,0 +1,225 @@
+package views.applicant;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.h3;
+import static j2html.TagCreator.label;
+
+import com.google.common.collect.ImmutableList;
+import controllers.applicant.routes;
+import j2html.TagCreator;
+import j2html.tags.specialized.ATag;
+import j2html.tags.specialized.ButtonTag;
+import j2html.tags.specialized.DivTag;
+import j2html.tags.specialized.FormTag;
+import j2html.tags.specialized.InputTag;
+import j2html.tags.specialized.LabelTag;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import play.i18n.Messages;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import services.Address;
+import services.MessageKey;
+import services.geo.AddressSuggestion;
+import services.geo.AddressSuggestionGroup;
+import views.ApplicationBaseView;
+import views.HtmlBundle;
+import views.components.Icons;
+import views.components.LinkElement;
+import views.style.ApplicantStyles;
+import views.style.StyleUtils;
+
+/** Renders a page indicating the applicant is not eligible for a program. */
+public final class AddressCorrectionBlockView extends ApplicationBaseView {
+  private final String BLOCK_FORM_ID = "cf-block-form";
+  private static int MAX_SUGGESTIONS_TO_DISPLAY = 3;
+  private final ApplicantLayout layout;
+  public static String USER_KEEPING_ADDRESS_VALUE = "USER_KEEPING_ADDRESS_VALUE";
+  public static String SELECTED_ADDRESS_NAME = "selectedAddress";
```

Comment:
**nitpick:** organize these to separate the static vars vs the ones set by the c'tor

Reply:
done